### PR TITLE
Implements a load-balancer following the "Power of Two Random Choices" strategy

### LINKED ISSUE #220: Provide a "Power of Two Random Choices" load-balancing strategy
See http://www.eecs.harvard.edu/~michaelm/NEWWORK/postscripts/twosurvey.pdf.

### PR DESCRIPTION
**Pull request by cescoffier**
Fix #220

Also:
- Use SecureRandom instead of Random
- Add a test in the static list discovery verifying that parsing errors are caught
- Add missing modules to the coverage
- Fix revapi package issues

### CODE REVIEW

**Review by aureamunoz — APPROVED** [review]
LGTM

File: load-balancer/power-of-two-choices/src/main/java/io/smallrye/stork/loadbalancer/poweroftwochoices/PowerOfTwoChoicesLoadBalancer.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+        List<ServiceInstance> instances = new ArrayList<>(serviceInstances);
+        int count = instances.size();
+
+        if (count == 1) {
+            return instances.get(0);
+        }
+
+        ServiceInstance first = instances.get(random.nextInt(count));
+        ServiceInstance second = instances.get(random.nextInt(count));
```
Comment (michalszynkiewicz) [inline load-balancer/power-of-two-choices/src/main/java/io/smallrye/stork/loadbalancer/poweroftwochoices/PowerOfTwoChoicesLoadBalancer.java:38]: shouldn't it select a different random instance?

Comment (michalszynkiewicz) [reply load-balancer/power-of-two-choices/src/main/java/io/smallrye/stork/loadbalancer/poweroftwochoices/PowerOfTwoChoicesLoadBalancer.java:38]: actually, we select these random two every time so the fact that we sometimes have the same two is not an issue

Comment (cescoffier) [reply load-balancer/power-of-two-choices/src/main/java/io/smallrye/stork/loadbalancer/poweroftwochoices/PowerOfTwoChoicesLoadBalancer.java:38]: yes, it's not a problem.
I was wondering to do an object equality check in this case, but it's clearly premature optimization.

**Pull request merged**